aspnetcore/fundamentals/routing/samples/8.x/OptionalParams/Controllers/MyController.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.AspNetCore.Mvc;
+
+namespace WebApplication1.Controllers;
+
+[Route("api/[controller]")]
+public class MyController : ControllerBase
+{
+    // GET /api/my/red/2/joe
+    // GET /api/my/red/2
+    // GET /api/my
+    [HttpGet("{color}/{id:int?}/{name?}")]
+    public IActionResult GetByIdAndOptionalName(string color, int id = 1, string? name = null)
```

Comment:
nit: As both `id` and `name` are optional, this method should probably be named `GetByOptionalIdAndName`. The route will match by requesting `color` alone